src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -221,12 +221,12 @@ public void OnResponseHeadersStart()
             public void OnResponse100Continue()
             {
                 // This is called when we receive complete set of headers with status 100 Continue.
-                // We need to rest state and wait for another final response.
+                // We need to reset state and wait for another final response.
                 lock (SyncObject)
                 {
                     if (_state != StreamState.ExpectingData)
                     {
-                        throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
+                        throw new Http2ProtocolException(SR.Format(SR.net_http_http2_protocol_state, "100Continue", _state));
```

Comment:
Is the string "100continue" actually received on the wire in the protocol.  Usually we see either a request header "Expect: 100-continue" or the response status line of "100 Continue".

Reply:
It is the second status code. I added space to match it.